olp-cpp-sdk-authentication/src/AuthenticationClient.cpp

```diff
@@ -450,206 +451,124 @@ AuthenticationClient::Impl::~Impl() { pending_requests_->CancelAllAndWait(); }
 client::CancellationToken AuthenticationClient::Impl::SignInClient(
     AuthenticationCredentials credentials, SignInProperties properties,
     AuthenticationClient::SignInClientCallback callback) {
-  if (!settings_.network_request_handler) {
-    ExecuteOrSchedule(settings_.task_scheduler, [callback] {
-      AuthenticationError result({static_cast<int>(http::ErrorCode::IO_ERROR),
-                                  "Cannot sign in while offline"});
-      callback(result);
-    });
-    return client::CancellationToken();
-  }
-  std::weak_ptr<http::Network> weak_network(settings_.network_request_handler);
+  auto sign_in_task =
+      [=](client::CancellationContext context) -> SignInClientResponse {
+    if (!settings_.network_request_handler) {
+      return {{static_cast<int>(http::ErrorCode::IO_ERROR),
+               "Cannot sign in while offline"}};
+    }
 
-  auto cache = client_token_cache_;
+    client::OlpClientSettings settings;
+    settings.network_request_handler = settings_.network_request_handler;
+    settings.proxy_settings = settings_.network_proxy_settings;
+    settings.retry_settings.backdown_strategy =
+        olp::client::ExponentialBackdownStrategy();
 
-  client::CancellationContext context;
+    client::OlpClient client;
+    client.SetBaseUrl(settings_.token_endpoint_url);
+    client.SetSettings(std::move(settings));
```

Comment:
Maybe move to an anonymous function?

Reply:
What do you mean about move to an anonymous function? All this code is already in sign_in_task function.

Reply:
I mean creating an instance of OlpClient helper function.